polkadot/node/core/pvf/src/testing.rs

```diff
@@ -62,6 +62,33 @@ pub fn validate_candidate(
 pub fn get_and_check_worker_paths() -> (PathBuf, PathBuf) {
 	// Only needs to be called once for the current process.
 	static WORKER_PATHS: OnceLock<Mutex<(PathBuf, PathBuf)>> = OnceLock::new();
+
+	fn maybe_build_workers(build_prep: bool, build_exec: bool) {
+		let mut build_args = vec!["build"];
+		if build_prep {
+			build_args.push("--bin=polkadot-prepare-worker");
+		}
+		if build_exec {
+			build_args.push("--bin=polkadot-execute-worker");
+		}
+
+		if build_prep || build_exec {
+			eprintln!("Building workers...");
+
+			let mut dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+			if !dir.ends_with("polkadot-sdk") {
+				dir.pop();
+			}
```

Comment:
```suggestion
			let dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
```

We have a workspace, you can run this command everywhere in the workspace and it works.

Reply:
The problem is we have so many of them and what `env!("CARGO_MANIFEST_DIR")` evaluates to depends on where you run `cargo test`, which doesn't necessarily have hose `--bin=*` build targets.

Reply:
I don't get what you are saying? `env!("CARGO_MANIFEST_DIR")` this is evaluated at compile time and thus, always return the same.

Reply:
Yeah, it stays the same for a specific crate/package. If you run `cargo test` in any sub-dir in `/polkadot-sdk/polkadot/node/core/pvf`, it would evaluate to `/polkadot-sdk/polkadot/node/core/pvf` because that's where the manifest of this package resides. Then the `cargo build` would break as the manifest doesn't define these build targets.

Reply:
```
error: no bin target named `polkadot-prepare-worker`.
```

You would get something like this if you run `cargo test`.

Reply:
`cargo build -p polkadot --bin=polkadot-prepare-worker`

works.

Reply:
Nice to know! This is more straightforward.